packages/block-library/src/index.native.js

```diff
@@ -130,20 +134,52 @@ const registerBlock = ( block ) => {
 	} );
 };
 
+/**
+ * Function to register a block variations e.g. social icons different types.
+ *
+ * @param {Object} block The block which variations will be registered.
+ *
+ */
+const registerBlockVariations = ( block ) => {
+	const { metadata, settings, name } = block;
+
+	settings.variations
+		.sort( ( a, b ) => {
+			if ( a.title < b.title ) {
+				return -1;
+			} else if ( a.title > b.title ) {
+				return 1;
+			}
+			return 0;
+		} )
+		.forEach( ( v ) => {
+			registerBlockType( `${ name }-${ v.name }`, {
+				...metadata,
+				name: `${ name }-${ v.name }`,
+				...settings,
+				icon: v.icon(),
+				title: v.title,
+				initialAttributes: {
+					service: v.name,
+				},
+			} );
+		} );
+};
+
 // only enable code block for development
 // eslint-disable-next-line no-undef
 const devOnly = ( block ) => ( !! __DEV__ ? block : null );
 
 const iOSOnly = ( block ) =>
 	Platform.OS === 'ios' ? block : devOnly( block );
 
-// Hide the Classic block
+// Hide the Classic block and SocialIcon block
 addFilter(
 	'blocks.registerBlockType',
 	'core/react-native-editor',
 	( settings, name ) => {
 		if (
-			name === 'core/freeform' &&
+			( name === 'core/freeform' || name === 'core/social-link' ) &&
```

Comment:
Hmmm, do you think that there will be more of these cases in the future? Maybe it's worth to create an array with them and check if it includes that string?

```js
const hiddenBlocks = [core/freeform', 'core/social-link'];
if( hiddenBlocks.includes( name ) && hasBlockSupport( settings, 'inserter', true ) ) {
  ...
}
```